### PR TITLE
Retry if device is not in IDLE state

### PR DESCRIPTION
**Pull request by vcdmp**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

Retry if device is not in IDLE state

### CODE REVIEW

File: plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c [hunk 1]
```diff
@@ -449,9 +450,19 @@ fu_logitech_bulkcontroller_device_write_firmware(FuDevice *device,
...
-		g_prefix_error(error, "error in writing init transfer packet: ");
+	/* Sending INIT. Retry if device is not in IDLE state to receive the file */
+	do {
+		if (!fu_logitech_bulkcontroller_device_send_upd_cmd(self, CMD_INIT, NULL, error)) {
+			g_prefix_error(error,
+				       "error in writing init transfer packet: retrying... ");
+			continue;
+		}
+		break;
+	} while (--init_retry);
```
Comment (hughsie) [inline plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c:461]: There are multiple problems with this, including re-using the GError and not setting the error on failure. Could you use `fu_device_retry()` instead? The idea with that is you create a callback (perhaps `fu_logitech_bulkcontroller_device_send_upd_cmd_cb`) that does the action, and then `fu_device_retry()` handles all the things like waiting for a set amount of time (say 20ms) and saving the GError in the appropriate way.

File: plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c [hunk 2]
```diff
@@ -165,7 +165,7 @@ fu_logitech_bulkcontroller_device_send(FuLogitechBulkcontrollerDevice *self,
 					(guint8 *)buf->data,
 					buf->len,
 					&transferred,
-					WRITE_TIME_OUT,
+					BULK_TRANSFER_TIMEOUT,
```
Comment (hughsie) [inline plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c:168]: this looks like it should be a different commit, no?

File: plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c (hunk 1 above)
Comment (vcdmp) [reply plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c]: Fixed

File: plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c (hunk 2 above)
Comment (vcdmp) [reply plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c]: Fixed.

**Comment by hughsie** [comment]
Can you rebase this against `main` please, and squash it down into one commit with the title prefix of `logitech-bulkcontroller: ` -- then LGTM, thanks.

**Comment by hughsie** [comment]
If it helps, what I do is:

    git checkout main
    git pull
    git checkout wip/vcdmp/notidle-retry
    git rebase main -i

squash down into one, and amend the first one, then

    git push --force

...to update the PR.

**Comment by vcdmp** [comment]
Not familiar with squash feature. Need help

**Comment by hughsie** [comment]
@vcdmp you've lost the chunk that does the retry.

**Comment by vcdmp** [comment]
Not familiar with squash feature, lost changes, closing this PR. Opened new PR: https://github.com/fwupd/fwupd/pull/3822

**Comment by hughsie** [comment]
> closing this PR. Opened new PR

That works too, thanks :)
